cli/awsbatch/awsbsub.py

```diff
@@ -249,6 +242,153 @@ def _upload_and_get_command(boto3_factory, args, job_name, region, s3_bucket, lo
     return command
 
 
+def _get_stdin_and_upload(boto3_factory, s3_bucket, job_folder, job_script, timeout):
+    """
+    Create file from STDIN and upload to S3.
+
+    :param boto3_factory: initialized Boto3ClientFactory object
+    :param s3_bucket: S3 bucket
+    :param job_folder: S3 bucket subfolder
+    :param job_script: job script name
+    :param timeout: S3 expiration time in seconds
+    """
+    try:
+        # copy stdin to temporary file
+        with os.fdopen(sys.stdin.fileno(), "rb") as src:
+            with open(job_script, "wb") as dst:
+                shutil.copyfileobj(src, dst)
+
+        _upload_to_s3(boto3_factory, s3_bucket, job_script, job_folder + job_script, timeout)
+    except (OSError, Exception) as e:
+        fail("Error creating job script. Failed with exception: %s" % e)
+    finally:
+        # remove temporary file
+        if os.path.exists(job_script):
+            os.remove(job_script)
+
+
+def _get_env_and_upload(boto3_factory, env, env_blacklist, s3_bucket, job_folder, env_file, timeout, log):
+    """
+    Get environment variables, create a file containing the list of the exported env variables and upload to S3.
+
+    :param boto3_factory: initialized Boto3ClientFactory object
+    :param env: comma separated list of environment variables
+    :param env_blacklist: comma separated list of blacklisted environment variables
+    :param s3_bucket: S3 bucket
+    :param job_folder: S3 bucket subfolder
+    :param env_file: environment file name
+    :param timeout: S3 expiration time in seconds
+    :param log: log
+    """
+    key_value_list = _get_env_key_value_list(env, log, env_blacklist)
+    try:
+        # copy env to temporary file
+        with open(env_file, "wb") as dst:
+            dst.write("\n".join(key_value_list) + "\n")
+
+        _upload_to_s3(boto3_factory, s3_bucket, env_file, job_folder + env_file, timeout)
+    except (OSError, Exception) as e:
+        fail("Error creating environment file. Failed with exception: %s" % e)
+    finally:
+        # remove temporary file
+        if os.path.exists(env_file):
+            os.remove(env_file)
+
+
+def _compose_bash_command(args, s3_bucket, region, job_folder, job_script, env_file):
+    """
+    Define bash command to execute.
+
+    :param args: input arguments
+    :param s3_bucket: S3 bucket
+    :param region: AWS region
+    :param job_folder: S3 bucket subfolder
+    :param job_script: job script file
+    :param env_file: environment file
+    :return: composed bash command
+    """
+    command_args = shell_join(args.arguments)
+    # download awscli, if required.
+    bash_command = (
```

Comment:
I was wondering if we should pipe the different commands with && rather than ; so that we break if a command fails